packages/core/test/unit/query-generator/create-table-query.test.ts

```diff
@@ -0,0 +1,342 @@
+import { DataTypes } from '@sequelize/core';
+import { expectsql, sequelize } from '../../support';
+
+// TODO: overhaul this test suite when migrating createTableQuery to TypeScript
+/**
+ * Make methods stricter, throw when using invalid options
+ * Accept and test for TableNameOrModel
+ * Check if createTableQuery is typed correctly
+ * Check if all tests make sense, the current tests are just copied from dialect specific tests and other expectations are added
+ * Give tests better names
+ * Make sure that all resulting queries are valid by adding integration tests for QueryInterface.createTable
+ */
+
+describe('QueryGenerator#createTableQuery', () => {
+  const queryGenerator = sequelize.getQueryInterface().queryGenerator;
+
+  it('produces a CREATE TABLE query with a single INTEGER attribute', () => {
+    expectsql(queryGenerator.createTableQuery('myTable', { int: 'INTEGER' }, {}), {
+      default: 'CREATE TABLE IF NOT EXISTS [myTable] ([int] INTEGER);',
+      'mariadb mysql': 'CREATE TABLE IF NOT EXISTS `myTable` (`int` INTEGER) ENGINE=InnoDB;',
+      mssql: `IF OBJECT_ID(N'[myTable]', 'U') IS NULL CREATE TABLE [myTable] ([int] INTEGER);`,
+      ibmi: `BEGIN DECLARE CONTINUE HANDLER FOR SQLSTATE VALUE '42710' BEGIN END; CREATE TABLE "myTable" ("int" INTEGER); END`,
+    });
+  });
+
+  it('produces a CREATE TABLE query with two VARCHAR attributes', () => {
+    expectsql(queryGenerator.createTableQuery('myTable', { title: 'VARCHAR(255)', name: 'VARCHAR(255)' }), {
+      default: 'CREATE TABLE IF NOT EXISTS [myTable] ([title] VARCHAR(255), [name] VARCHAR(255));',
+      'mariadb mysql': 'CREATE TABLE IF NOT EXISTS `myTable` (`title` VARCHAR(255), `name` VARCHAR(255)) ENGINE=InnoDB;',
+      mssql: `IF OBJECT_ID(N'[myTable]', 'U') IS NULL CREATE TABLE [myTable] ([title] VARCHAR(255), [name] VARCHAR(255));`,
+      ibmi: `BEGIN DECLARE CONTINUE HANDLER FOR SQLSTATE VALUE '42710' BEGIN END; CREATE TABLE "myTable" ("title" VARCHAR(255), "name" VARCHAR(255)); END`,
+    });
+  });
+
+  it('produces a CREATE TABLE query with two VARCHAR attributes and a tableName object', () => {
```

Comment:
I don't think the fact that two varchars attributes is relevant to this test and could be simplified to just one, what's important seems to be testing that TableName is used

Also:

```suggestion
  it('produces a CREATE TABLE query with two VARCHAR attributes and a TableNameWithSchema object', () => {
```